src/lib/api/inline.preboot.code.ts

```diff
@@ -104,31 +104,76 @@ export function getEventRecorderCode(): string {
 }
 
+/**
+ * Used by the server side version of preboot. The main purpose
+ * is to get the inline code that can be inserted into the server view.
+ * Returns the definitions of the prebootInit function called in code returned by
+ * getInlinePrebootCodeInvocation.
+ *
+ * @param customOptions PrebootRecordOptions that override the defaults
+ * @returns Generated inline preboot code with just functions definitions
+ * to be used separately
+ */
+export function getInlinePrebootCodeDefinition(customOptions?: PrebootOptions): string {
+  const opts = <PrebootOptions>assign({}, defaultOptions, customOptions);
+
+  // safety check to make sure options passed in are valid
+  validateOptions(opts);
+
+  const scriptCode = getEventRecorderCode();
+
+  // wrap inline preboot code with a self executing function in order to create scope
+  const initStr = init.toString();
+  return `var ${initFunctionName} = (function() {
+      ${scriptCode}
+      return (${initStr.replace('common_1.', '')});
+    })();`;
+}
+
 
 /**
  * Used by the server side version of preboot. The main purpose
  * is to get the inline code that can be inserted into the server view.
+ * Invokes the prebootInit function defined in getInlinePrebootCodeDefinition with proper
+ * parameters. Each appRoot should get a separate inlined code from a separate call to
+ * getInlinePrebootCodeInvocation but only one inlined code from getInlinePrebootCodeDefinition.
  *
  * @param customOptions PrebootRecordOptions that override the defaults
- * @returns Generated inline preboot code is returned
+ * @returns Generated inline preboot code with just invocations of functions from
+ * getInlinePrebootCodeDefinition
  */
-export function getInlinePrebootCode(customOptions?: PrebootOptions): string {
+export function getInlinePrebootCodeInvocation(customOptions?: PrebootOptions): string {
   const opts = <PrebootOptions>assign({}, defaultOptions, customOptions);
 
   // safety check to make sure options passed in are valid
   validateOptions(opts);
 
   const optsStr = stringifyWithFunctions(opts);
-  const scriptCode = getEventRecorderCode();
 
-  // TODO re-add minification option?
+  return `${initFunctionName}(${optsStr});`;
+}
+
+
+/**
+ * Used by the server side version of preboot. The main purpose
+ * is to get the inline code that can be inserted into the server view.
+ * Returns both definitions of functions needed by Preboot & their invocations.
+ * Works best when there is only one appRoot as it needs to be injected for each
+ * of them separately so with multiple appRoots it generates lots of code.
+ * Preboot's Angular integration no longer uses this function but
+ * getInlinePrebootCodeDefinition and getInlinePrebootCodeInvocation instead.
+ *
+ * @param customOptions PrebootRecordOptions that override the defaults
+ * @returns Generated inline preboot code is returned
+ */
+export function getInlinePrebootCode(customOptions?: PrebootOptions): string {
```

Comment:
The issues here are that if someone uses `getInlinePrebootCode` directly:
1. They need to move it from `<head>` to just after the opening app root tag which may not be easy in all scenarios. This is a breaking change.
2. If multiple app roots are used, each of them will need to get this injected separetely which increases the HTML.

Should we still support this function or just go all the way & require people to use `getInlinePrebootCodeDefinition` & `getInlinePrebootCodeInvocation` separately?

Reply:
Yeah let's cut it. I'll confirm with @jeffwhelpley offline, but so long as we document this change _very_ well, I don't foresee it being a huge problem.

Is there any way we can streamline the process with a utility for React users?

Reply:
I don't know React well enough so I can't speak for it. How is it used there? AFAIK React has DOM rehydration so there should be no separate server & client node, should it?

Reply:
Sorry I haven't given comments yet on this PR. I have been slammed getting ready for ng-conf. 

So, we did have a decent number of React users before they introduced hydration. It probably isn't as popular any more, but just realize that hydration doesn't solve all the problems. For example, if a user clicks on a button before the client is active, nothing is going to happen.

Regardless, the number of people is likely really small. I suggest we just move forward and we can always add some sugar later for non-Angular users if people complain.